src/main/java/command/Commands.java

```diff
@@ -0,0 +1,351 @@
+     * @throws DukeException The exception thrown when encountering any problems in executing.
+     */
+    public int execute(ListOfTask taskList, int lineNumber, String error) throws DukeException {
+        boolean load = true;
+        if (lineNumber == 0 && error == null) {
+            load = false;
+        }
+            switch (this.state) {
+
+            case BYE:
+                return 0;
+
+            case LIST:
+                taskList.listTasks();
+                break;
+
+            case TODO:
+                if (!load) {
+                    taskList.addTask(this.name, true);
+                } else {
+                    taskList.addTask(this.name, false);
+                }
+                break;
+
+            case FIND:
+                taskList.find(this.name);
+                break;
+
+            case SORT:
+                break;
+
+            case MARK:
+                if (!load) {
+                    taskList.mark(this.index, true);
+                } else {
+                    taskList.mark(this.index, false);
+                }
+                break;
+
+            case UNMARK:
+                if (!load) {
+                    taskList.unMark(this.index, true);
+                } else {
+                    taskList.unMark(this.index, false);
+                }
+                break;
+
+            case DELETE:
+                if (!load) {
+                    taskList.delete(this.index, true);
+                } else {
+                    taskList.delete(this.index, false);
+                }
+                break;
+            }
+        return 1;
+    }
+
+    /**
+     * Compares LocalDateTime between this object and c.
+     *
+     * @param c Commands object to be compared to.
+     * @return Returns true if this object's time is before c's time, false for all other cases.
+     */
+    public boolean compareTime(Commands c) {
+        try {
+            if (this.dateTime.isBefore(c.dateTime)) {
+                return true;
+            }
+        } catch (NullPointerException e) {
+            return false;
+        }
+        return false;
+    }
+
+    /**
+     * Compares this object to another object.
+     *
+     * @param obj An object.
+     * @return Returns true if both objects are equivalent, false if otherwise.
+     */
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj instanceof Commands) {
+            Commands b = (Commands) obj;
+            if (b.state == this.state
+                    && b.dateTime == this.dateTime
+                    && b.index == this.index
+                    && b.name == this.name) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static class TwoCommands extends Commands {
+        private COMMANDS state2;

```

Comment:
Could there be a better name for this variable? A more relatable variable name would be better!